src/core/lib/security/transport/secure_endpoint.cc

```diff
@@ -91,9 +107,13 @@ struct secure_endpoint {
   /* saved handshaker leftover data to unprotect. */
   grpc_slice_buffer leftover_bytes;
   /* buffers for read and write */
-  grpc_slice read_staging_buffer = GRPC_SLICE_MALLOC(STAGING_BUFFER_SIZE);
-  grpc_slice write_staging_buffer = GRPC_SLICE_MALLOC(STAGING_BUFFER_SIZE);
+  grpc_slice read_staging_buffer ABSL_GUARDED_BY(read_mu) = grpc_empty_slice();
+  grpc_slice write_staging_buffer ABSL_GUARDED_BY(write_mu) =
+      grpc_empty_slice();
   grpc_slice_buffer output_buffer;
+  grpc_core::MemoryOwner memory_owner;
+  grpc_core::MemoryAllocator::Reservation self_reservation;
+  bool has_posted_reclaimer = false;
```

Comment:
what's the thread synchronization story for this? seems like it's accessed by both the reader and the writer, in which case we probably want to make it a std::atomic<bool> and use an exchange in post_reclaimer

Reply:
Yes it's accessed by both - changed to atomic.